main.cpp

```diff
@@ -6,25 +6,62 @@
 
 #include "xos.h"
 
-int main(int argc, char **argv) {
-  if (argc < 2) {
+int main(int argc, char **argv)
+{
+  if (argc < 2)
+  {
     std::cerr << "Missing input file" << std::endl;
     return 1;
   }
 
   std::ifstream input(argv[1]);
   xos::Lexer lexer(input);
-  while (1) {
-    xos::Result<xos::Token> tok = lexer.getNextToken();
-    if (tok.hasError()) {
+  xos::Result<xos::Token> tok{xos::Token::eof, 0, 0, xos::Token::val{.eof = true}}; // set to a random value -- for now for testing.
```

Comment:
This `tok` doesn't need to be outside the loop below since it's only used inside that loop. So you can move it there and have:

```
  while (1) {
    auto tok = lexer.getNextToken();
```